### DISCUSSION TITLE
[Feature] DolphinScheduler Lineage

### DISCUSSION THREAD

**Discussion by zhongjiajie** [apache/dolphinscheduler#6596] [Ideas]
This discussion is about DolphinScheduler Lineage. The ref ISSUE is https://github.com/apache/dolphinscheduler/issues/249, but it more like vote tickle instead of issue, So I create a discussion to continue the work.

I init thinks we should draw following conclusions after discuss:

* The purpose of data lineage(from https://github.com/apache/dolphinscheduler/discussions/6596#discussioncomment-1535823)
  *  Discover potential dependencies
  *  Build workflow through data lineage automation
  * Easy to debug
* The way we implement lineage, we have to add a documentation before we start write the code(Of cause we could add some POC code for it)
* To be adding if someone have other idea.

**Comment by wen-hemin** [comment]
It is recommended to start with collecting requirements, What is the purpose of data lineage ?

- Discover potential dependencies
- Build workflow through data lineage automation
- Easy to debug

**Reply by zhongjiajie** [reply]
Good point 👍 , I think all those are the purpose of data lineage, but we should discuss the necessity and the difficulty of them. And make roadmap to implement.

**Reply by ChenpiDog** [reply]
Data lineage display function is very important for data warehouse, and it is the responsibility of task scheduling tool.

**Reply by ChenpiDog** [reply]
DolphinScheduler can first support the Data lineage of SQL tasks (Spark SQL, Flink SQL, SQL, etc.). Many users urgently need this function.

**Reply by ChenpiDog** [reply]
@zhongjiajie 
![DolphinScheduler数据血缘沟通（2022-10-02） drawio](https://user-images.githubusercontent.com/55483982/193460165-46e67d8c-9005-46f7-881f-075898fbf6cc.png)

**Reply by ChenpiDog** [reply]
The data blood relationship information is not suitable for storage in traditional relational databases. Whether to consider introducing a graph database to store this kind of information.

**Reply by zhongjiajie** [reply]
@ChenpiDog looking good, do you interested in this feature? I think currently this issue still in help wantted

**Reply by ChenpiDog** [reply]
@zhongjiajie I'm very interested in this, but I'm not familiar with Java and can't contribute to the code. I can provide some ideas.

**Reply by ChenpiDog** [reply]
There are two main aspects to consider in visualization:
  1) Automatically generate Lineage data kinship from online code (such as SQL);
  2) It is necessary to consider the views and materialized views of the database, which are generally stored in the database meta information table.

Note: The global Lineage data kinship needs to combine the above two points. For example, the source table used in my SQL is a view.
